build/media-suite_helper.sh

```diff
@@ -429,19 +429,17 @@ do_extract() {
     local archive="$1" dirName="$2"
     # accepted: zip, 7z, tar, tar.gz, tar.bz2 and tar.xz
     [[ -z $dirName ]] && dirName=$(guess_dirname "$archive")
-    if [[ $dirName != "." && -d $dirName ]] &&
-        {
-            [[ $build32 == "yes" ]] && { [[ ! -f "$dirName"/build_successful32bit ]] ||
-                [[ -n $flavor && ! -f "$dirName/build_successful32bit_${flavor}" ]]; } ||
-                [[ $build64 == "yes" ]] && { [[ ! -f "$dirName"/build_successful64bit ]] ||
-                    [[ -n $flavor && ! -f "$dirName/build_successful64bit_${flavor}" ]]; }
-        }; then
-        rm -rf "$dirName"
-    elif [[ -d $dirName ]]; then
+    if [[ $dirName != "." && -d $dirName ]]; then
+        if [[ $build32 == "yes" ]] &&
+            [[ ! -f "$dirName"/build_successful32bit || -n $flavor && ! -f "$dirName/build_successful32bit_${flavor}" ]]; then
+            rm -rf "$dirName"
+        elif [[ $build64 == "yes" ]] &&
+            [[ ! -f "$dirName"/build_successful64bit || -n $flavor && ! -f "$dirName/build_successful64bit_${flavor}" ]]; then
+            rm -rf "$dirName"
+        fi
+    elif [[ -d $dirName ]] || ! expr "$archive" : '.\+\(tar\(\.\(gz\|bz2\|xz\)\)\?\|7z\|zip\)$' > /dev/null; then
```

Comment:
The expr part does not do the same thing. If it's not an archive, it should not try to enter a non-existing dirName